src/python/pants/backend/codegen/protobuf/protobuf_dependency_inference.py

```diff
@@ -41,22 +40,22 @@ class ProtobufMapping:
 
 @rule(desc="Creating map of Protobuf file names to Protobuf targets", level=LogLevel.DEBUG)
 async def map_protobuf_files(protobuf_targets: AllProtobufTargets) -> ProtobufMapping:
-    stripped_sources_per_target = await MultiGet(
-        Get(StrippedSourceFileNames, SourcesPathsRequest(tgt[ProtobufSourceField]))
+    stripped_file_per_target = await MultiGet(
+        Get(
+            StrippedFileName, StrippedFileNameRequest(cast(str, tgt[ProtobufSourceField].file_path))
```

Comment:
To make type hints more useful for `SingleSourceField`, I think it makes sense to keep `SingleSourceField` and add `OptionalSingleSourceField` too, which allows 0-1 sources. We use both a lot now, and the type system doesn't reflect this nuance. Callers have to do things like `cast`.

Reply:
Will do this in a separate PR. I can land that first if we want less churn in this file